modules/apps/captcha/captcha-test/src/testIntegration/java/com/liferay/captcha/internal/settings/test/CaptchaSettingsImplTest.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.captcha.internal.settings.test;
+
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.captcha.configuration.CaptchaConfiguration;
+import com.liferay.captcha.provider.CaptchaProvider;
+import com.liferay.portal.configuration.test.util.CompanyConfigurationTemporarySwapper;
+import com.liferay.portal.kernel.captcha.CaptchaSettings;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.util.HashMapDictionaryBuilder;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Pedro Victor Silvestre
+ */
+@RunWith(Arquillian.class)
+public class CaptchaSettingsImplTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Test
+	public void test() throws Exception {
+		try (CompanyConfigurationTemporarySwapper
+				companyConfigurationTemporarySwapper =
+					new CompanyConfigurationTemporarySwapper(
+						TestPropsValues.getCompanyId(),
+						CaptchaConfiguration.class.getName(),
+						new HashMapDictionaryBuilder(
+						).<String, Object>put(
+							"createAccountCaptchaEnabled", false
+						).build())) {
+
+			CaptchaConfiguration captchaConfiguration =
+				_captchaProvider.getCaptchaConfiguration();
+
+			Assert.assertEquals(
+				captchaConfiguration.createAccountCaptchaEnabled(),
```

Comment:
Hey @daiazius, Can you add a case to verify the behavior with more than one company?  

PS: It may be similar to this test one here `com.liferay.captcha.configuration.test.CaptchaConfigurationTest`

Thanks